CMakeLists.txt

```
@@ -105,9 +105,12 @@ find_package(OpenGL REQUIRED)
 
 set(CMAKE_AUTOMOC ON)
 
-find_package(Qt5 REQUIRED COMPONENTS Core Widgets OpenGL)
-set(QT_LIBRARIES Qt5::Widgets)
-set(QTVERSION ${Qt5Widgets_VERSION})
+find_package(QT NAMES Qt6 REQUIRED COMPONENTS Core)
+# in theory: find_package(QT NAMES Qt6 Qt5 REQUIRED COMPONENTS Core)
```

I would expect that _all_ components are available from the same Qt version:
```suggestion
find_package(QT NAMES Qt6 Qt5 REQUIRED COMPONENTS Core Widgets OpenGL)
```

---

In theory my version would fail if there is Qt6 Core installed but not OpenGL, yours would fallback to Qt5 if that is more complete.

In practice somehow it always picks Qt5 even if Qt6 is available

---

https://github.com/isys-vision/rviz/pull/2

---

Your version successfully picked Qt6 here when available, thanks! Maybe the `NAMES` needs a certain CMake version to work? I took it from the Qt docs ...